packages/toolpad-app/src/toolpadDataSources/local/client.tsx

```diff
@@ -113,74 +129,99 @@ function QueryEditor({
+                    label={name}
+                    propType={definiton}
+                    jsRuntime={jsBrowserRuntime}
+                    renderControl={(renderControlParams) => (
+                      <Control {...renderControlParams} propType={definiton} />
+                    )}
+                    value={paramsObject[name]}
+                    onChange={(newValue) => {
+                      const paramKeys = Object.keys(parameterDefs);
+                      const newParams: BindableAttrEntries = paramKeys.flatMap((key) => {
+                        const paramValue = key === name ? newValue : paramsObject[key];
+                        return paramValue ? [[key, paramValue]] : [];
+                      });
+                      setInput((existing) => ({
+                        ...existing,
+                        params: newParams,
+                      }));
+                    }}
+                  />
+                ) : null;
+              })}
+            </Stack>
           </Stack>
-          <Typography>Parameters:</Typography>
-          <Stack gap={1}>
-            {Object.entries(parameterDefs).map(([name, definiton]) => {
-              const Control = getDefaultControl(definiton, liveBindings);
-              return Control ? (
-                <BindableEditor
-                  key={name}
-                  liveBinding={liveBindings[name]}
-                  globalScope={globalScope}
-                  globalScopeMeta={globalScopeMeta}
-                  label={name}
-                  propType={definiton}
-                  jsRuntime={jsBrowserRuntime}
-                  renderControl={(renderControlParams) => (
-                    <Control {...renderControlParams} propType={definiton} />
-                  )}
-                  value={paramsObject[name]}
-                  onChange={(newValue) => {
-                    const paramKeys = Object.keys(parameterDefs);
-                    const newParams: BindableAttrEntries = paramKeys.flatMap((key) => {
-                      const paramValue = key === name ? newValue : paramsObject[key];
-                      return paramValue ? [[key, paramValue]] : [];
-                    });
-                    setInput((existing) => ({
-                      ...existing,
-                      params: newParams,
-                    }));
-                  }}
-                />
-              ) : null;
-            })}
-          </Stack>
-        </Stack>
-      </QueryInputPanel>
-
-      <QueryPreview isLoading={previewIsLoading} error={preview?.error}>
-        <JsonView sx={{ height: '100%' }} copyToClipboard src={preview?.data} />
-      </QueryPreview>
-    </SplitPane>
+        </QueryInputPanel>
+
+        <QueryPreview isLoading={previewIsLoading} error={preview?.error}>
+          <JsonView sx={{ height: '100%' }} copyToClipboard src={preview?.data} />
+        </QueryPreview>
+      </SplitPane>
+
+      <Dialog
```

Comment:
You'll have to extract and reuse the logic for https://github.com/mui/mui-toolpad/blob/b083a524b91dc35eb2572412b05aa8abb4930a97/packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/CreateCodeComponentNodeDialog.tsx#L117C1-L117C1

Reply:
Agreed, we should have a reusable Button which allows opening the editor. I'll refactor this to allow reuse and some more configurability